authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt

```diff
@@ -0,0 +1,193 @@
+package com.bitwarden.authenticatorbridge.manager
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.os.IBinder
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ProcessLifecycleOwner
+import com.bitwarden.authenticatorbridge.IAuthenticatorBridgeService
+import com.bitwarden.authenticatorbridge.manager.model.AccountSyncState
+import com.bitwarden.authenticatorbridge.manager.model.AuthenticatorBridgeConnectionType
+import com.bitwarden.authenticatorbridge.manager.util.toPackageName
+import com.bitwarden.authenticatorbridge.model.EncryptedSharedAccountData
+import com.bitwarden.authenticatorbridge.provider.AuthenticatorBridgeCallbackProvider
+import com.bitwarden.authenticatorbridge.provider.StubAuthenticatorBridgeCallbackProvider
+import com.bitwarden.authenticatorbridge.provider.SymmetricKeyStorageProvider
+import com.bitwarden.authenticatorbridge.util.decrypt
+import com.bitwarden.authenticatorbridge.util.toFingerprint
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+private const val AUTHENTICATOR_BRIDGE_SERVICE_CLASS =
+    "com.x8bit.bitwarden.data.platform.service.AuthenticatorBridgeService"
+
+/**
+ * Default implementation of [AuthenticatorBridgeManager].
+ *
+ * @param applicationContext The Context that will be used to bind to AuthenticatorBridgeService. Should be an
+ * application context.
+ * @param connectionType Specifies which build variant to connect to.
+ * @param symmetricKeyStorageProvider Provides access to local storage of the symmetric encryption
+ * key.
+ * @param callbackProvider Provides a way to construct a service callback that can be mocked in
+ * tests.
+ * @param processLifecycleOwner Lifecycle owner that is used to listen for start/stop
+ * lifecycle events.
+ */
+class AuthenticatorBridgeManagerImpl(
+    private val applicationContext: Context,
+    private val connectionType: AuthenticatorBridgeConnectionType,
+    private val symmetricKeyStorageProvider: SymmetricKeyStorageProvider,
+    callbackProvider: AuthenticatorBridgeCallbackProvider = StubAuthenticatorBridgeCallbackProvider(),
+    processLifecycleOwner: LifecycleOwner = ProcessLifecycleOwner.get(),
+) : AuthenticatorBridgeManager, DefaultLifecycleObserver {
```

Comment:
We can remove `DefaultLifecycleObserver` here right?

Reply:
Oh yeah: 9d789e6a4fbe204d334da38e06aa57a1baa911d1